README.md

```diff
@@ -29,95 +33,319 @@ are well established for RSE-ops, and which require further development. We can
+The end-goal of an HPC environment is to make software available to researchers. This is abstractly similar to the "end goal" of a cloud native environment, which is to make software available to users via services. For cloud native, I suspect this typically means interactive web interfaces or services that are accessed from other web or terminal clients via application programming interfaces (APIs). Since HPC environments are typically running jobs that rely on system software and some number of known aliases, the bulk majority of "software distribution" for HPC is probably command line interfaces. In this framework, the administrators must provide an easy way to manage a collection of software, and make it available to users.
+
+> Again here it's not just admins. App teams maintain their own app deployments. The end result of
+> that is that there are directories in the filesystem with apps and their dependencies -- and
+> there are potentially modules as well so you can `module load` specific apps/versions. Most
+> deployments right now are to shared filesystem directories (`/project` dirs at NERSC,
+> `/usr/gapps` at LLNL, etc. and maintained by app teams (often very specific people on the app
+> teams)
+
 Users are somewhat empowered to install their own software, given that 1. they have the filesystem space to do so, and 2. installation does not require any escalated permissions on the system.
-While this could mean a "bring your own environment" scenario via containers pulled from external registries, more likely it means relying on package managers and module systems. Ideally, every module and package would have a suite of tests to ensure its functionality, but in real world scenarios its likely that the software is tested elsewhere (e.g., a CI service associated with the source code) and distributed to the system for immediate usage.
+While this could mean a "bring your own environment" scenario via containers pulled from external registries, more likely it means relying on package managers and module systems.
+
+> With better HPC container runtimes this could get fixed.
+
+Ideally, every module and package would have a suite of tests to ensure its functionality, but in real world scenarios its likely that the software is tested elsewhere (e.g., a CI service associated with the source code) and distributed to the system for immediate usage.
+
+> Depends on external software vs. internal. This stuff is very very true for external software --
+> OSS may not be tested directly on our machines (though we have stuff like spack test now -- not
+> everything has tests).
+
+> Internal codes are tested a lot -- there are devlops teams and dedicated clusters for them. Codes
```

Comment:
Yeah that's pretty great! But there is nothing like this in academic HPC centers.

Reply:
that's true but is this document only for academic HPC centers?  I don't think it should be.

Reply:
Also see above.  There very much are large academic codes that are long-lived and have developers working on them.  It depends on the school/project/funding/etc.  We fund a lot of these via, e.g., PSAAP.  UINTAH at Utah, ADCIRC at UNC, NAMD at UIUC, lots of astro codes, various quantum codes... etc.  It's not all small stuff in academia.